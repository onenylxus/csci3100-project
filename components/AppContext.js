// Import
import React from 'react';

// Export authentication context
export default React.createContext();
