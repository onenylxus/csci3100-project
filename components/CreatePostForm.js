// Import
import React from 'react';
import { TextInput, View } from 'react-native';
import Style from '../assets/style';

// Export Create Post Form
export default function CreatePostForm() {
  // const [post, setPost] = React.useState('');

  return (
    <View style={Style.inputContainer}>
      <TextInput
        style={Style.createPostBox}
        multiline
        scrollEnabled
        enablesReturnKeyAutomatically
        placeholder="What's on your mind?"
        // onChangeText={(text) => setPost(text)}
      />
    </View>
  );
}
