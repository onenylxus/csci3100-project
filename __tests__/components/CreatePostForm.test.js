// Import
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';

// Import target component
import AppContext from '../../components/AppContext';
import CreatePostForm from '../../components/CreatePostForm';

// Mock authentication method
const AppMethodMock = {
  login: jest.fn(),
  logout: jest.fn(),
  getUser: jest.fn(),
};

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Mock navigation route
jest.mock('@react-navigation/core');

// Run jest tests
describe('CreatePostForm', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(
      <NavigationContainer>
        <AppContext.Provider value={AppMethodMock}>
          <CreatePostForm />
        </AppContext.Provider>
      </NavigationContainer>
    );
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
