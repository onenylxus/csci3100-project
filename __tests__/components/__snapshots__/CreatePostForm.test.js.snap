// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePostForm matches snapshot 1`] = `null`;
